syntax = "proto3";

option java_package = "com.vell.chat.account";

package account;

// 账号服务
service Account {
  rpc signup (SignupMsg) returns (AccountResp) {}
  rpc login (LoginMsg) returns (AccountResp) {}
  rpc logout (TokenMsg) returns (AccountResp) {}
  rpc isAlive (TokenMsg) returns (AccountResp) {}
}

// 注册信息
message SignupMsg {
  string username = 1; // 用户名，唯一存在
  string password = 2;
  string phoneNumber = 3;
  string email = 4;
  string extra = 5;
}

// 登陆信息
message LoginMsg {
  string account = 1; // 账号，可以是用户名、手机号、email
  string password = 2;
}

// token信息
message TokenMsg {
  string token = 1;
  int32 expiration_time_sec = 2;
}

// 账号结果返回
message AccountResp {
  int32 code = 1; // 错误码
  string msg = 2;
  TokenMsg token = 3;
  string extra = 4;
}